### PR TITLE
[travis] Kill tests when a new commit has been pushed

### PR DESCRIPTION
**Pull request by nicolas-grekas**
| Q | A |
| --- | --- |
| Bug fix? | no |
| New feature? | no |
| BC breaks? | no |
| Deprecations? | no |
| Tests pass? | yes |
| Fixed tickets | - |
| License | MIT |
| Doc PR | - |

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -26,6 +26,8 @@ env:
         - SYMFONY_DEPRECATIONS_HELPER=weak
 
 before_install:
+    - if [[ "$TRAVIS_PULL_REQUEST" != "false" ]]; then git fetch origin "refs/pull/$TRAVIS_PULL_REQUEST/head"; else git fetch origin "$TRAVIS_BRANCH"; fi;
+    - if [[ "$TRAVIS_COMMIT" != `git rev-parse FETCH_HEAD` ]]; then exit 1; fi;
```
Comment (stof) [inline .travis.yml]: you should probbaly output a message explaining why this has exited

Comment (nicolas-grekas) [inline .travis.yml]: I'm not much inspired, is it ok for you now?

**Comment by stof** [comment]
:+1:

**Comment by nicolas-grekas** [comment]
I tried to find a way to "cancel" the build but it looks like it's not possible, so we must `exit 1` which puts the build in "errored" mode.

**Comment by fabpot** [comment]
Thank you @nicolas-grekas.

**Pull request merged**